### PR TITLE
126 Style Sort By Dropdown

### PR DESCRIPTION
**Pull request by EthanGu4**
### Description
[//]: # "[Required] Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Added the dropdown feature for the sort by button
- Added styling for the dropdown bar and components within the dropdown bar
- No functionality added atm

### Screenshots
[//]: # "Add any screenshots that will help your PR reviewer understand what’s going on in the PR!"
[//]: # "[Required] For frontend tasks: add screenshots of the layout and expected behavior (GIFs if you’re feeling extra fancy)!"
[//]: # "Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

https://github.com/user-attachments/assets/eb9e2177-128b-43f8-8699-f84cb7a95eec

## How to review
[//]: # "/[Required] Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"
- Review components/ProjectsListingModal/index.tsx first for the creation of the dropdown bar and the items contained within the dropdown bar, as well as logic for the open/close icon
- Review styling in components/ProjectsListingModal/styles.ts to see if the current state of the sort by dropdown bar matches the design in the figma

## Next steps
[//]: # "[Optional] - What has not yet been added to the PR, doesn’t yet have any functionality — essentially, what needs to be fixed or added."
[//]: # "[Optional] any temporary fixes in this PR that should be cleaned up later."
- Probably functionality? Also there were some minor issues with how the dropdown bar looked but I think we found a fix? (just adjusting it to be aligned toward the left)

### Online sources
[//]: # "[Optional] insert links to any useful tutorials or documentation that you used for this PR!"

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @jjstnlee

### CODE REVIEW

File: components/ProjectsListingModal/index.tsx [hunk 1]
```diff
@@ -44,6 +46,9 @@ export default function ProjectsListingModal({
   const closeModal = () => setIsModalOpen(false);
   const openModal = () => setIsModalOpen(true);
 
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
```
Comment (jjstnlee) [inline components/ProjectsListingModal/index.tsx:49]: pls rename this to isSortByOpen to make it more clear what kind of dropdown it is

File: components/ProjectsListingModal/index.tsx [hunk 2]
```diff
@@ -90,10 +95,34 @@ export default function ProjectsListingModal({
                     <GlobeIcon width={'0.5625rem'} height={'0.5625rem'} />
                     <SubHeading2>ALL PROJECTS</SubHeading2>
                   </AllProjectsHeader>
-                  <SortByButton onClick={() => {}}>
-                    <SubHeading2>SORT BY</SubHeading2>
-                    <OpenIcon width={'10'} height={'14'} />
-                  </SortByButton>
+                  <div style={{ position: 'relative' }}>
```
Comment (jjstnlee) [inline components/ProjectsListingModal/index.tsx:98]: pls create a styled component in styles.ts called SortBy (can be a styled.div)

File: components/ProjectsListingModal/styles.ts [hunk 3]
```diff
@@ -70,18 +70,63 @@ export const ProjectItemsDiv = styled.div`
...
+  border-right: 0.5px solid rgba(46, 58, 89, 0.08);
+  border-bottom: 1.5px solid rgba(46, 58, 89, 0.08);
+  border-left: 0.5px solid rgba(46, 58, 89, 0.08);
+  background: #fafafb;
+  box-shadow: 0px -2px 5px 0px #fff;
+  z-index: 3;
+  right: 0;
+`;
+
+export const DropdownItem = styled.button`
```
Comment (jjstnlee) [inline components/ProjectsListingModal/styles.ts:111]: rename DropdownMenu and DropdownItem to SortByMenu and SortByItem

**Comment by jjstnlee** [comment]
pls create a new text thing in styles/text.ts for the Sort By Items font. follow the examples above in the file.

**Review by ethan-tam33 — COMMENTED** [review]
I agree with Julee's recommended revisions. Besides those, LGTM!
